pandas/core/indexes/multi.py

```diff
@@ -89,6 +89,9 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
             raise TypeError("Must pass both levels and labels")
         if len(levels) != len(labels):
             raise ValueError('Length of levels and labels must be the same.')
+        if names is not None and len(names) > 0 and len(levels) == 0:
+            levels = [[] for _ in names]
+            labels = [[] for _ in names]
```

Comment:
I added the code here to ensure all creation paths had the same behavior. Otherwise the same checks and empty array creation code would need to be added to both `.from_tuples` and `.from_array`.